internal/services/applicationinsights/application_insights_resource.go

```diff
@@ -414,6 +408,141 @@ func resourceApplicationInsightsRead(d *pluginsdk.ResourceData, meta interface{}
 	return nil
 }
 
+func resourceApplicationInsightsUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).AppInsights.ComponentsClient
+	billingClient := meta.(*clients.Client).AppInsights.BillingClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := components.ParseComponentID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.ComponentsGet(ctx, *id)
+	if err != nil {
+		if !response.WasNotFound(existing.HttpResponse) {
+			return fmt.Errorf("checking for presence of existing %s: %v", id, err)
+		}
```

Comment:
We can update this to just return the error message regardless of whether we received a 404 or not

```suggestion
	return fmt.Errorf("retrieving %s: %v", id, err)
```